src/moonlink/src/row/column_array_builder.rs

```diff
@@ -95,9 +97,90 @@ impl ColumnArrayBuilder {
                 array_builder,
             ),
             DataType::List(inner) => ColumnArrayBuilder::new(inner.data_type(), capacity, true),
+            DataType::Struct(fields) => ColumnArrayBuilder::Struct(
+                StructBuilder::from_fields(fields.clone(), capacity),
+                array_builder,
+            ),
             _ => panic!("data type: {data_type:?}"),
         }
     }
+
+    fn append_value_to_struct_field(
+        builder: &mut StructBuilder,
+        i: usize,
+        value: &RowValue,
+    ) -> Result<(), Error> {
+        if let Some(bool_builder) = builder.field_builder::<BooleanBuilder>(i) {
+            match value {
+                RowValue::Bool(v) => bool_builder.append_value(*v),
+                RowValue::Null => bool_builder.append_null(),
+                _ => unreachable!("Bool expected from well-typed input"),
```

Comment:
I might be too paranoid, I would add logging for failed matching values, otherwise we have no way to debug, see https://github.com/Mooncake-Labs/moonlink/pull/1256